### PR TITLE
React | Assignment | Added the first assignment

### PR DESCRIPTION
**Pull request by anandchavan24**

### CODE REVIEW

File: src/App.tsx [hunk 1]
```diff
@@ -1,24 +1,27 @@
...
 import './App.css';
 
+import { useState } from 'react';
+import TodoList from './Components/ToDoList';
+import { ToDoInterface } from './Shared/interface';
+import { sampleDataForList } from './Shared/sample';
+
 function App() {
+  const toDoArray:ToDoInterface[] = sampleDataForList;
+  const [todos, setTodos] = useState(toDoArray);
```
Comment (Rushikesh-josh) [inline src/App.tsx:10]: define type for state

File: src/App.tsx [hunk 2]
```diff
@@ -1,24 +1,27 @@
...
+
+  const markCompleted = (id:number) => {
+    setTodos((prevTodos:ToDoInterface[]) =>
```
Comment (Rushikesh-josh) [inline src/App.tsx]: don't use the interface key in the name you can use ITodos.
it will indicate interface of todos

File: src/App.tsx [hunk 3]
```diff
@@ -1,24 +1,27 @@
...
+      prevTodos.map((todo:ToDoInterface) => (todo.id === id ? { ...todo, completed: !todo.completed } : todo))
+    );
+  };
+
+  const deleteTodo = (id:number) => {
+    setTodos((prevTodos:ToDoInterface[]) => prevTodos.filter((todo) => todo.id !== id));
```
Comment (Rushikesh-josh) [inline src/App.tsx]: is previous state required here? I mean (prevTodos:ToDoInterface[])  is this required?

File: src/Components/ToDo.tsx [hunk 4]
```diff
@@ -0,0 +1,21 @@
+import { ToDoProps } from "../Shared/interface";
```
Comment (Rushikesh-josh) [inline src/Components/ToDo.tsx]: instead of interface use types name for the file
"../Shared/interface"

File: src/Shared/sample.ts
Comment (Rushikesh-josh) [inline src/Shared/sample.ts:1]: please define json instead of .ts

File: src/App.tsx (hunk 3 above)
Comment (anandchavan24) [reply src/App.tsx]: Yes

Here we need to maintain previous state while  updating the data

File: src/Components/ToDo.tsx (hunk 4 above)
Comment (anandchavan24) [reply src/Components/ToDo.tsx]: Done

File: src/App.tsx (hunk 3 above)
Comment (Rushikesh-josh) [reply src/App.tsx]: can you try with 
setTodos(( todos.filter((todo) => todo.id !== id));

Comment (anandchavan24) [reply src/App.tsx]: Yes it works as expected.

I willl note this point

File: src/Shared/sample.ts
Comment (anandchavan24) [reply src/Shared/sample.ts:1]: Sure.

I will define nested JSON as discussed

**Review by Rushikesh-josh — APPROVED** [review]

**Review by Rushikesh-josh — COMMENTED** [review]
Is this second assignment? fetching data using the custom hook?
If yes please raise separate MR once your merge first assignment code into main

**Pull request merged**